Weather App/ViewModel/CurrentWeatherViewModel.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  CurrentWeatherViewModel.swift
+//  Weather App
+//
+//  Created by Shikhar Sharma on 22/03/22.
+//
+
+import Foundation
+import UIKit
+import CoreLocation
+
+public class WeatherViewModel {
+    let locationName = Box("Loading..")
+    let currentWeatherDescription = Box(CurrentWeatherCompact())
+    let currentAirQuality = Box(AirQualityCompact())
+    let currentWindDescription = Box(WindCompact())
+    var forecastedData = Box([DayForecastCompact]())
+    var backGroundImage = Box(UIImage())
+    var hourforecastToday = Box([HourForecastCompact]())
+    var astroToday = Box(AstroCompact())
+    
+    private static func roundTwoDecimalLabel (value: Double) -> String {
+        let val = round(value * 100) / 100.0
+        let ans = String(format: "%g", val)
+        return ans
+    }
+    
+    public static var defaultAddress = "New Delhi"
+    
+    func checkCondition(conditionText : String) -> String {
+        if conditionText.contains("snow") || conditionText.contains("ice") ||
+            conditionText.contains("blizzard") ||
+            conditionText.contains("Blizzard"){
+            return "snow"
+        }
+        if conditionText.contains("rain") || conditionText.contains("drizzle") || conditionText.contains("sleet"){
+            return "rain"
+        }
+        if conditionText.contains("thunder") {
+            return "thunder"
+        }
+        if conditionText.contains("mist") ||
+            conditionText.contains("fog") {
+            return "mist"
+        }
+        if conditionText.contains("sunny") ||
+            conditionText.contains("clear") ||
+            conditionText.contains("cloud") || conditionText.contains("overcast"){
+            return "clear"
+        }
+        return "default"
+    }
+    
+    func getBackgroundImage(isDay: Int, conditionText: String) -> UIImage {
+        let condition = conditionText.lowercased()
+        var conditionName = checkCondition(conditionText: condition)
+        if conditionName == "default" || conditionName == "thunder" {
+            return UIImage(named: conditionName)!
+        }
+        if isDay == 1 {
+            conditionName += "Day"
+        } else {
+            conditionName += "Night"
+        }
+        return UIImage(named: conditionName)!
+    }
+    
+    func setCurrentLocation () {
+        let locManager = CLLocationManager()
+        locManager.requestWhenInUseAuthorization()
+        var currentLocation: CLLocation!
+
+        if
+           CLLocationManager.authorizationStatus() == .authorizedWhenInUse ||
+           CLLocationManager.authorizationStatus() ==  .authorizedAlways
+        {
+            currentLocation = locManager.location
+            changeLocation(to: "\(currentLocation.coordinate.latitude), \(currentLocation.coordinate.longitude)", isAqi: true)
+        }
+    }
+    
+    func changeLocation (to city: String, isAqi: Bool) {
```

Comment:
isAqi signifies a bool wether we want to fetch data for Air Quality index also, we can modify this but this is how it was given in api, so we went further to avoid much difference

Reply:
put isAqi as true by default and removed from parameter. Done